### PR TITLE
Feature/month view calendar

### PR DESCRIPTION
**Pull request by DavidMartel12**
## 📖 Description
I added the monthly view to the new calendar view.

### 🧪 How Has This Been Tested?
Existing tests have been modified to implement the new information required for the calendar view.

### ☑️ Checklist before requesting a review
- [x] I have performed a self-review of my code.
- [x] If it is a core feature, I have added thorough tests.
- [x] If needed, I added analytics.
- [x] Make sure to add either one of the following labels: `version: Major`,`version: Minor` or `version: Patch`. 
- [x] Make sure golden files changes were reviewed and approved.

### 🖼️ Screenshots (if useful):
Screenshot_1694552060

### CODE REVIEW

File: lib/core/viewmodels/schedule_viewmodel.dart [hunk 1]
```diff
@@ -155,6 +157,18 @@ class ScheduleViewModel extends FutureViewModel<List<CourseActivity>> {
     return events;
   }
 
+  List<CalendarEventData> selectedMonthCalendarEvents() {
+    final List<CalendarEventData> events = [];
+    final date = selectedDate.datesOfMonths();
+    for (int i = 0; i < 42; i++) {
```
Comment (camillebrulotte) [inline lib/core/viewmodels/schedule_viewmodel.dart:163]: Peut-être mettre le 42 dans une constante pour savoir ça représente quoi ? :D

**Review by camillebrulotte — CHANGES_REQUESTED** [review]

File: lib/ui/views/schedule_view.dart [hunk 2]
```diff
@@ -175,6 +179,44 @@ class _ScheduleViewState extends State<ScheduleView>
     final chevronColor = Theme.of(context).brightness == Brightness.light
         ? AppTheme.primaryDark
         : AppTheme.lightThemeBackground;
+
+    model.handleViewChanged(DateTime.now(), eventController);
+
+    if (model.calendarFormat == CalendarFormat.month) {
+      return Scaffold(
```
Comment (XavierPaquet-Rapold) [inline lib/ui/views/schedule_view.dart:186]: Est-ce qu'on pourrait mettre ce Scaffold dans une classe différente pour que ce soit plus un Component ?

**Review by XavierPaquet-Rapold — APPROVED** [review]
LGTM après l'update du pubspec

**Review by camillebrulotte — APPROVED** [review]
LGTM !

**Review by XavierPaquet-Rapold — DISMISSED** [review]

**Review by clubapplets-server — DISMISSED** [review]
LGTM! -Sam

**Review by MysticFragilist — DISMISSED** [review]

**Review by XavierPaquet-Rapold — DISMISSED** [review]

**Review by XavierPaquet-Rapold — APPROVED** [review]

**Review by camillebrulotte — APPROVED** [review]
LGTM !

**Pull request merged**